docs/porting/target/mpu.md

```diff
@@ -0,0 +1,46 @@
+<h2 id="mpu-port">MPU</h2>
+
+Implementing the MPU API enables Mbed OS to provide better security by preventing code execution and code modification where is shouldn't be allowed.
+
+### Assumptions
+
+#### Defined behavior
+
+- The function `mbed_mpu_init` is safe to call repeatedly
+- The function `mbed_mpu_free` disables MPU protection
+- Execution from RAM results in a fault when execute never is enabled. This RAM includes heap, stack, data and zero init
+- Writing to ROM results in a fault when write never is enabled
+
+#### Undefined behavior
+
+- Calling any function other than `mbed_mpu_init` before the initialization of the MPU.
+
+### Dependencies
+
+Hardware MPU capabilities.
+
+### Implementing the MPU API
+
+You can find the API and specification for the MPU API in the following header file:
+
+[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/group__hal__mpu.html)
+
+To enable MPU support in Mbed OS, add the `MPU` label in the `device_has` option of the target's section in the `targets.json` file.
+
+Targets with a standard ARMv7-M or ARMv8-M MPU, indicated by `__MPU_PRESENT` being defined to 1 in the target's CMSIS header, only need the `MPU` label in the `device_has` for MPU support. This pulls in a common Mbed OS MPU driver, so in this case a target specific driver is not needed.
+
+Targets with a standard ARMv7-M or ARMv8-M MPU needing to override the common Mbed OS MPU driver can do so by defining `MBED_MPU_CUSTOM`. This removes the common Mbed OS MPU driver from builds so a target specific one can be used instead.
+
+Targets with a standard ARMv7-M or ARMv8-M MPU with the common Mbed OS MPU driver can customize where flash ends and ram begins by defining `MBED_MPU_ROM_END`. By default this is set to `0x0FFFFFFF` for  ARMv7-M and `0x1FFFFFFF` for ARMv8-M. The address space up to and including this is treated as rom which the MPU protects from write accesses. The address space after this is treated as ram which the MPU protects from code execution.
```

Comment:
This paragraph needs to be removed since a separate mechanism will be used for this